istioctl/pkg/rbac/upgrade.go

```diff
@@ -97,56 +75,47 @@ func (ug *Upgrader) upgradeLocalFile(fileName string) (string, error) {
 			return "", err
 		}
 		role := ug.upgradeServiceRole(serviceRole)
-		err = writeToFile(newFile, role)
+		convertedRole, err := parseConfigToString(role)
 		if err != nil {
 			return "", err
 		}
+		convertedPolicies.WriteString(convertedRole)
 	}
 	for _, serviceRoleBinding := range ug.serviceRoleBindings {
 		authzPolicy := ug.createAuthorizationPolicyFromRoleBinding(serviceRoleBinding)
-		err = writeToFile(newFile, authzPolicy)
+		convertedAuthzPolicy, err := parseConfigToString(authzPolicy)
 		if err != nil {
 			return "", err
 		}
+		convertedPolicies.WriteString(convertedAuthzPolicy)
 	}
-	fmt.Println(fmt.Sprintf("New authorization policies have been upgraded to %s", newFilePath))
-	return newFilePath, nil
+	return convertedPolicies.String(), nil
 }
 
-// writeToFile writes data from `config` to `file`.
-func writeToFile(file *os.File, config model.Config) error {
-	var stringBuilder strings.Builder
+// parseConfigToString parses data from `config` to string.
+func parseConfigToString(config model.Config) (string, error) {
 	schema, exists := configDescriptor.GetByType(config.Type)
 	if !exists {
-		return fmt.Errorf("unknown kind %q for %v", crd.ResourceName(config.Type), config.Name)
+		return "", fmt.Errorf("unknown kind %q for %v", crd.ResourceName(config.Type), config.Name)
 	}
 	obj, err := crd.ConvertConfig(schema, config)
 	if err != nil {
-		return fmt.Errorf("could not decode %v: %v", config.Name, err)
+		return "", fmt.Errorf("could not decode %v: %v", config.Name, err)
 	}
-	stringBuilder.WriteString(fmt.Sprintf("%s%s/%s\n", apiVersionField, config.Group, config.Version))
-	stringBuilder.WriteString(fmt.Sprintf("%s%s\n", kindField, obj.GetObjectKind().GroupVersionKind().Kind))
-	stringBuilder.WriteString(fmt.Sprintf("%s\n", metadataField))
-	stringBuilder.WriteString(fmt.Sprintf("  %s%s\n", nameField, config.Name))
-	stringBuilder.WriteString(fmt.Sprintf("  %s%s\n", namespaceField, config.Namespace))
-	stringBuilder.WriteString(fmt.Sprintf("%s\n", specField))
-	spec, err := yaml.Marshal(obj.GetSpec())
+	configInBytes, err := yaml.Marshal(obj)
 	if err != nil {
-		return err
-	}
-	specLines := strings.Split(string(spec), "\n")
-	for i, line := range specLines {
-		// If specLines has an empty line at the end, ignore it for better formatting.
-		if i == len(specLines)-1 && line == "" {
-			break
+		return "", fmt.Errorf("could not unmarshal %v: %v", config.Name, err)
+	}
+	configLines := strings.Split(string(configInBytes), "\n")
+	var configInString strings.Builder
+	for i, configLine := range configLines {
+		if i == len(configLines)-1 && configLine == "" {
+			configInString.WriteString("---\n")
+		} else if !strings.Contains(configLine, creationTimestampField) {
```

Comment:
`yaml.Marshal(obj)` outputs `creationTimestamp: nil` field for all policies. `convert_ingress.go` also has this, but I don't think they handle this case, so I manually check and ignore this field. Everything else looks good @yangminzhu